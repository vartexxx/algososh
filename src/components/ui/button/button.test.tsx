import renderer, {ReactTestRendererJSON} from 'react-test-renderer';
import {Button} from './button';
import {fireEvent, render, screen} from '@testing-library/react';
import Mock = jest.Mock;


describe('Тестирование компонента Button', (): void => {
    test('Компонент Button корректно рендерится с текстом', (): void => {
        const buttonText: string = 'Нажми меня';
        const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer.create(<Button text={buttonText} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Компонент Button корректно рендерится без текста', (): void => {
        const buttonText: string = '';
        const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer.create(<Button text={buttonText} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Компонент Button корректно рендерится в заблокированном состоянии', (): void => {
        const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer.create(<Button disabled={true} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Компонент Button корректно рендерится с индикатором загрузки', (): void => {
        const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer.create(<Button isLoader={true} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Компонент Button корректно отрабатывает callback', (): void => {
        const onClick: Mock<any, any> = jest.fn();
        const buttonText: string = 'Нажми меня';
        render(<Button text={buttonText} onClick={onClick} />);
        const link: HTMLElement = screen.getByText('Нажми меня');
        fireEvent.click(link);
        expect(onClick).toHaveBeenCalledTimes(1);
    })
})