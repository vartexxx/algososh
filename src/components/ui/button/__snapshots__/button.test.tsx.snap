// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Button Компонент Button корректно рендерится без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button Компонент Button корректно рендерится в заблокированном состоянии 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button Компонент Button корректно рендерится с индикатором загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирование компонента Button Компонент Button корректно рендерится с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Нажми меня
  </p>
</button>
`;
